src/main/java/week2/BTree.java

```diff
@@ -0,0 +1,113 @@
+package week2;
+
+
+class BTree<T extends Comparable<T>> {
+
+  private boolean bothFound = false;
+  private Node<T> root;
```

Comment:
functionality updated to have a method that returns an array of ancestor and this is used by both the printAncestors and lowestCommonAncestor